spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/BaseSqsIntegrationTest.java

```diff
@@ -0,0 +1,72 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs;
+
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.SQS;
+
+import com.amazonaws.auth.AWSCredentials;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeAll;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+
+@Testcontainers
+abstract class BaseSqsIntegrationTest {
+
+	protected static final String RECEIVES_MESSAGE_QUEUE_NAME = "receives.message.test.queue";
+	protected static final String DOES_NOT_ACK_ON_ERROR_QUEUE_NAME = "does.not.ack.test.queue";
+	protected static final String RESOLVES_PARAMETER_TYPES_QUEUE_NAME = "resolves.parameter.test.queue";
+	protected static final String RESOLVES_POJO_TYPES_QUEUE_NAME = "resolves.pojo.test.queue";
+	protected static final String RECEIVE_FROM_MANY_1_QUEUE_NAME = "receive.many.test.queue.1";
+	protected static final String RECEIVE_FROM_MANY_2_QUEUE_NAME = "receive.many.test.queue.2";
+	protected static final String ASYNC_RECEIVE_FROM_MANY_1_QUEUE_NAME = "async.receive.many.test.queue.1";
+	protected static final String ASYNC_RECEIVE_FROM_MANY_2_QUEUE_NAME = "async.receive.many.test.queue.2";
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(SQS).withReuse(true);
+
+	static StaticCredentialsProvider credentialsProvider;
+
+	@BeforeAll
+	static void beforeAll() throws IOException, InterruptedException {
+		// create needed queues in SQS
+		// TODO: Not working as expected due to some port mapping issue - will look into in the future
+		localstack.execInContainer("awslocal", "io/awspring/cloud/sqs", "create-queue", "--queue-name",
+				RECEIVES_MESSAGE_QUEUE_NAME);
+		localstack.execInContainer("awslocal", "io/awspring/cloud/sqs", "create-queue", "--queue-name",
+				DOES_NOT_ACK_ON_ERROR_QUEUE_NAME);
+		localstack.execInContainer("awslocal", "io/awspring/cloud/sqs", "create-queue", "--queue-name",
+				RECEIVE_FROM_MANY_1_QUEUE_NAME);
```

Comment:
For initialization resources we can use `docker-entrypoint-initaws.d`
> v0.9.0: Enhance integration with Serverless; refactor CloudFormation implementation; add support for Step Functions, IAM, STS; fix CloudFormation integration; support mounting Lambda code locally; add docker-entrypoint-initaws.d dir for initializing resources; add S3Event Parser for Lambda; fix S3 chunk encoding; fix S3 multipart upload notification; add dotnetcore2.1 and ruby2.5 Lambda runtimes; fix issues with JDK 9; install ES plugins available in AWS

From [changelog](https://github.com/localstack/localstack/blob/3d8b2835c72a3a2dbd2dd7b9be4d8dfd6c529873/CHANGELOG.md)

Reply:
But then you can't use constants or any other variables there. As long as the number of commands is short, for me its easier to have them in the same file. I think we can start using `docker-entrypoint-initaws.d` but then it should be consistent across tests - so perhaps something for a separate PR.